### PR TITLE
[FIX] bug showing "undefined" as channel name in Room Info actions Modal

### LINKED ISSUE #23759: [BUG] Channel name shown as "undefined" in the Room Information Actions Modal
### Description:

 Few but not all of the channel names are shown as "undefined" in the warning modals for the actions such as leave, hide etc.

### Steps to reproduce:

1.  Open the General Channel, Go to Room Information in the Contextual Bar at the top.
2.  Click on leave or hide the room.
3.  The warning modal shows the room name as "undefined".

### Expected behavior:

 Channel name should be displayed instead of "undefined".

### Actual behavior:

https://user-images.githubusercontent.com/59637086/142714670-5b70ce73-e598-4c0a-97e8-2bdb2f49c8a5.mp4

### Server Setup Information:

- Version of Rocket.Chat Server: 4.2.0 - develop
- Operating System: WSL-2 (Windows)
- Deployment Method:  Manual installation
- Number of Running Instances: 1
- DB Replicaset Oplog: Enabled
- NodeJS Version: 12.22.1 -x64
- MongoDB Version: 4.4.4 wiredTiger

### Client Setup Information

- Desktop App or Browser Version: Chrome latest Version 96.0.4664.45
- Operating System: Windows 10 version 21H1

### PR DESCRIPTION
**Pull request by kulkarnigaurav38**
- [X] I have read the Contributing Guide
  - [X] I have signed the CLA - https://cla-assistant.io/RocketChat/Rocket.Chat
  - [X] Lint and unit tests pass locally with my changes
  - [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
  - [ ] I have added necessary documentation (if applicable)
  - [ ] Any dependent changes have been merged and published in downstream modules

### BEFORE
https://user-images.githubusercontent.com/59637086/142716598-d45a4054-63ff-40e8-8b3f-bb11b6994142.mp4

### AFTER
https://user-images.githubusercontent.com/59637086/142716500-64a6a686-97d5-4725-b894-6acf15387ef6.mp4

## Issue(s)

Closes #23759 
## Steps to test or reproduce
 Mentioned in the issue.

### CODE REVIEW

**Review by ujorgeleite — APPROVED** [review]

**Review by ggazzo — CHANGES_REQUESTED** [review]
Please the check your solution with discussions and multiple DMs, I'm afraid it will not work as expected.

File: apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js [hunk 1]
```diff
@@ -46,7 +46,7 @@ const RoomInfoWithData = ({ rid, openEditing, onClickBack, onEnterRoom, resetSta
 	room.type = room.t;
 	room.rid = rid;
 
-	const { type, fname, prid, joined = true } = room; // TODO implement joined
+	const { type, name, prid, joined = true } = room; // TODO implement joined
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js:49]:
```suggestion
	const { type, fname, name, prid, joined = true } = room; // TODO implement joined
```

File: apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js [hunk 2]
```diff
@@ -117,7 +117,7 @@ const RoomInfoWithData = ({ rid, openEditing, onClickBack, onEnterRoom, resetSta
 
 		setModal(
 			<WarningModal
-				text={t(warnText, fname)}
+				text={t(warnText, name)}
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js:120]:
```suggestion
				text={t(warnText, fname | name)}
```

File: apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js [hunk 3]
```diff
@@ -142,7 +142,7 @@ const RoomInfoWithData = ({ rid, openEditing, onClickBack, onEnterRoom, resetSta
...
```
Comment (dougfabris) [inline apps/meteor/client/views/room/contextualBar/Info/RoomInfo/RoomInfoWithData.js:145]:
```suggestion
				text={t(warnText, fname | name)}
```

**Review by dougfabris — CHANGES_REQUESTED** [review]

**Comment by dougfabris** [comment]
Closing this one in favor of an older solution #20459
Thanks a lot for helping us @kulkarnigaurav38

**Pull request closed without merging**